Resolve dynamic imports of bare module specifiers at runtime

Seems that bare module specifiers aren't resolved at runtime.

Example

```typescript
try {
  await import("bare_module_specifier");
} catch (err) {
  console.log(err);
}
```

Results in the following compile time error

```bash
error: TS2307 [ERROR]: Cannot find module 'bare_module_specifier' or its corresponding type declarations.
 await import("bare_module_specifier");
```

This should be resolved at runtime, that's kind of the point of dynamic imports with top level await.

--- a/cli/module_graph.rs
+++ b/cli/module_graph.rs
@@ -1960,6 +1960,24 @@ pub mod tests {
   }
 
   #[tokio::test]
+  async fn test_graph_check_ignores_dynamic_import_errors() {
+    let specifier =
+      ModuleSpecifier::resolve_url_or_path("file:///tests/dynamicimport.ts")
+        .expect("could not resolve module");
+    let (graph, _) = setup(specifier).await;
+    let result_info = graph
+      .check(CheckOptions {
+        debug: false,
+        emit: false,
+        lib: TypeLib::DenoWindow,
+        maybe_config_path: None,
+        reload: false,
+      })
+      .expect("should have checked");
+    assert!(result_info.diagnostics.is_empty());
+  }
+
+  #[tokio::test]
   async fn fix_graph_check_emit_diagnostics() {
     let specifier =
       ModuleSpecifier::resolve_url_or_path("file:///tests/diag.ts")

--- a/cli/tests/module_graph/file_tests-dynamicimport.ts
+++ b/cli/tests/module_graph/file_tests-dynamicimport.ts
@@ -1,0 +1,5 @@
+try {
+  await import("bare_module_specifier");
+} catch (err) {
+  console.log(err);
+}

